packages/mdc-segmented-button/segmented-button/foundation.ts

```diff
@@ -40,27 +40,38 @@ export class MDCSegmentedButtonFoundation extends MDCFoundation<MDCSegmentedButt
     super({...MDCSegmentedButtonFoundation.defaultAdapter, ...adapter});
   }
 
-  selectSegment(_indexOrSegmentId: number | string) {
-    return;
+  selectSegment(indexOrSegmentId: number | string) {
+    this.adapter.selectSegment(indexOrSegmentId);
   }
 
-  unselectSegment(_indexOrSegmentId: number | string) {
-    return;
+  unselectSegment(indexOrSegmentId: number | string) {
+    this.adapter.unselectSegment(indexOrSegmentId);
   }
 
   getSelectedSegments(): readonly SegmentDetail[] {
-    return [];
+    return this.adapter.getSegments().filter(segmentDetail => segmentDetail.selected);
   }
 
-  isSegmentSelected(_indexOrSegmentId: number | string): boolean {
-    return false;
+  isSegmentSelected(indexOrSegmentId: number | string): boolean {
+    let segment = this.adapter.getSegments().filter(segmentDetail => segmentDetail.index === indexOrSegmentId || segmentDetail.segmentId === indexOrSegmentId);
+    return segment.length > 0 && segment[0].selected;
   }
 
   isSingleSelect(): boolean {
-    return false;
+    return this.adapter.hasClass('mdc-segmented-button--single-select');
   }
 
-  handleSelected(_detail: SegmentDetail) {
-    return;
+  handleSelected(detail: SegmentDetail) {
+    if (this.isSingleSelect()) {
+      let selectedSegments = this.getSelectedSegments();
+      if (detail.selected) {
+        selectedSegments.filter(segmentDetail => segmentDetail.index !== detail.index)
```

Comment:
please use `for of` loop as `filter(...).forEach(...)` creates an unnecessary array